src/components/ec-text-field/ec-text-field.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="ec-text-field">
+    <div
+      v-if="label"
+      class="ec-text-field__label"
+    >{{ label }}</div>
+    <input
+      class="ec-text-field__input"
+      :class="{
+        'ec-text-field__input--error': state === 'error',
+      }"
+      :type="type"
+      :value="value"
+      @input="(e) => $emit('change', e.target.value)"
+    >
+    <div
+      v-if="state === 'error'"
+      class="ec-text-field__error-label"
+    >{{ errorMessage }}</div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'EcTextField',
+  model: {
+    prop: 'value',
+    event: 'change',
+  },
+  props: {
+    type: {
+      type: String,
+      default: 'text',
+      validator(value) {
+        return ['text', 'date', 'number'].includes(value);
```

Comment:
Why the component is called a "text field" if you can pass date and number? Either remove this prop or (more preferably) change  the name of the component to `EcInput` or `EcInputField`.